src/raft_log.rs

```diff
@@ -379,19 +379,24 @@ impl<T: Storage> RaftLog<T> {
     }
 
     /// Returns entries starting from a particular index and not exceeding a bytesize.
-    pub fn entries(&self, idx: u64, max_size: impl Into<Option<u64>>) -> Result<Vec<Entry>> {
+    pub fn entries(
+        &self,
+        idx: u64,
+        max_size: impl Into<Option<u64>>,
+        context: GetEntriesContext,
+    ) -> Result<Vec<Entry>> {
         let max_size = max_size.into();
         let last = self.last_index();
         if idx > last {
             return Ok(Vec::new());
         }
-        self.slice(idx, last + 1, max_size)
+        self.slice(idx, last + 1, max_size, context)
     }
 
     /// Returns all the entries.
     pub fn all_entries(&self) -> Vec<Entry> {
         let first_index = self.first_index();
-        match self.entries(first_index, None) {
+        match self.entries(first_index, None, GetEntriesContext::Test) {
```

Comment:
This may imply `all_entries` must only use in test, how about passing `GetEntriesContext::Test` as a parameter to `all_entries` instead.

Reply:
But seems `all_entries` is indeed only used by tests.

Reply:
Yes, but nothing can stop people use it outside of tests.

Reply:
Or if this function is dedicated to test, it may be better to update the comment or change the name to something like `all_entries_test`

Reply:
will update the comment